modules/controller/index.js

```diff
@@ -37,193 +159,91 @@ Controller.prototype._relations = function (request) {
   if (result && !Array.isArray(result)) {
     result = [result];
   }
-  return this.validRelations(_.uniq(result));
+  return _.intersection(this.source.relations(), _.uniq(result));
 };
 
-Controller.prototype.validFilters = function (filters) {
-  return _.intersection(this.source.filters(), filters);
-};
+Controller.prototype._readRelatedHandler =
+  function(opts, request, response, next) {
+    var source = this.source;
+    var relation = request.params ? request.params.relation : null;
+    var responder = opts.responder || this.responder;
 
-Controller.prototype.validRelations = function(relations) {
-  return _.intersection(this.source.relations(), relations);
-};
 
-Controller.prototype.responder = require('./lib/responder');
+    source.byId(request.params.id, relation).then(function (model) {
```

Comment:
I was just coming over here to say...  :smile: 

We only interact with a facade over the source, and that can be promisified. Dropping the callbacks is what really opened everything up and let me see what could be abstracted. That said, if promisifying facades is a pain, it would be less gross to go back to callbacks using your PR as a starting point.